### PR TITLE
Permit 32-bit GCC compilation

### PR DESCRIPTION
**Pull request by knapply**
Following up on #976

### CODE REVIEW

**Comment by lemire** [comment]
@knapply  I have issued a PR on your PR at https://github.com/knapply/simdjson/pull/1

Please have a look.

**Comment by lemire** [comment]
I have added tests there: https://github.com/simdjson/simdjson/pull/992

(Obviously, they will fail, but I can always sync the branch later.)

File: src/westmere/bitmanipulation.h [hunk 1]
```diff
@@ -66,7 +66,7 @@ really_inline bool add_overflow(uint64_t value1, uint64_t value2,
 #endif
 }
 
-#ifdef SIMDJSON_REGULAR_VISUAL_STUDIO
+#if defined(SIMDJSON_REGULAR_VISUAL_STUDIO) || defined(SIMDJSON_IS_32BITS)
```
Comment (lemire) [inline src/westmere/bitmanipulation.h:69]: My expectation is that a 32-bit compiler would never make it there. I don't the code change, but I'd like to check my expectation with you.

File: src/haswell/bitmanipulation.h [hunk 2]
```diff
@@ -58,7 +58,7 @@ really_inline bool add_overflow(uint64_t value1, uint64_t value2,
 #endif
 }
 
-#ifdef SIMDJSON_REGULAR_VISUAL_STUDIO
+#if defined(SIMDJSON_REGULAR_VISUAL_STUDIO) || defined(SIMDJSON_IS_32BITS)
```
Comment (lemire) [inline src/haswell/bitmanipulation.h:61]: (Redundant comment.) My expectation is that a 32-bit compiler would never make it there. I don't the code change, but I'd like to check my expectation with you.

**Comment by lemire** [comment]
I think that this is essentially correct, but I'd like @knapply  to consider my PR and to review my question(s).

Cheers!

**Comment by lemire** [comment]
I am marking this 0.4.1. 

That would the final step for a 0.4.1 release... but I'd like to have tests matching.

**Comment by lemire** [comment]
@knapply Oh! And can you state that you verified that it actually solves the downstream issue? I'd like to be sure before we issue a release, because I don't think I know how to test under mingw. I am fine not have it in CI if I can get 32-bit GCC under Linux, but I'd still like for someone to say that they tested it and that it works.

(We will not support mingw directly, without CI, obviously.)

File: src/westmere/bitmanipulation.h (hunk 1 above)
Comment (knapply) [reply src/westmere/bitmanipulation.h:69]: I manually checked several times on all of these changes. 32-bit GCC (both  "vanilla" Linux, and the MinGW Windows port) make it here.

File: src/haswell/bitmanipulation.h (hunk 2 above)
Comment (knapply) [reply src/haswell/bitmanipulation.h:61]: (Redundant) I manually checked several times on all of these changes. 32-bit GCC (both  "vanilla" Linux, and the MinGW Windows port) make it here.

**Comment by knapply** [comment]
Yes, this solves the issue. I have 32-bit GCC tests in progress on a branch from my fork via CircleCI (I knew it would take a few attempts). I will to this PR as soon as I confirm they're good to go.

**Comment by knapply** [comment]
> I have added tests there: #992
> 
> (Obviously, they will fail, but I can always sync the branch later.)

Should I skip adding the Circle CI tests for GCC 7/10 (32-bit) that I'm verifying? 

GCC 10 works: https://app.circleci.com/pipelines/github/knapply/simdjson/7/workflows/99e2068d-4677-4fa5-8e64-7ee41859295e

Waiting on the GCC 7 addition to finish: https://app.circleci.com/pipelines/github/knapply/simdjson/8/workflows/cdf9317d-cf99-44c7-9618-e6ef53a5d4c8

**Comment by lemire** [comment]
> Should I skip adding the Circle CI tests for GCC 7/10 (32-bit) that I'm verifying?

No please add them.

**Comment by knapply** [comment]
I spoke too soon. The tests weren't obeying `-m32` and a dozen attempts later and I'm not making any progress. I need to do some more digging.

If you'd like to wait for me to sort that out first, that's totally fair. I just don't think I'll get a chance in the next day or two.

**Comment by lemire** [comment]
That's fine. I will probably merge, fix a line or two, and then see if I can add my own tests on top of it. If it looks good, I will release 0.4.1.

**Pull request merged**